plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+        ) as fileDialog:
+            if fileDialog.ShowModal() == wx.ID_CANCEL:
+                return  # the user changed their mind
+
+            # save the current contents in the file
+            pathname = fileDialog.GetPath()
+            if not pathname.endswith(".csv"):
+                pathname += ".csv"
+            try:
+                with open(pathname, "w", newline="") as csvfile:
+                    data = {}
+                    writer = csv.DictWriter(csvfile, fieldnames=CSVFieldnames.list())
+                    writer.writeheader()
+                    project_ctrls = self._project_list_controller.project_ctrls
+                    for project_ctrl in project_ctrls.values():
+                        data[CSVFieldnames.PROJECT_NAME.value] = project_ctrl.model.name.value
+                        roas = project_ctrl.model.roas.value
+                        for roa in roas:
+                            data[CSVFieldnames.ROA_NAME.value] = roa.name.value
+                            roa_coordinates = roa.coordinates.value
+                            data[CSVFieldnames.ROA_COORDINATES_L.value] = roa_coordinates[0]
+                            data[CSVFieldnames.ROA_COORDINATES_T.value] = roa_coordinates[1]
+                            data[CSVFieldnames.ROA_COORDINATES_R.value] = roa_coordinates[2]
+                            data[CSVFieldnames.ROA_COORDINATES_B.value] = roa_coordinates[3]
+                            roc_2 = roa.roc_2.value
+                            data[CSVFieldnames.ROC_2_NAME.value] = roc_2.name.value
+                            roc_2_coordinates = roc_2.coordinates.value
+                            data[CSVFieldnames.ROC_2_COORDINATES_L.value] = roc_2_coordinates[0]
+                            data[CSVFieldnames.ROC_2_COORDINATES_T.value] = roc_2_coordinates[1]
+                            data[CSVFieldnames.ROC_2_COORDINATES_R.value] = roc_2_coordinates[2]
+                            data[CSVFieldnames.ROC_2_COORDINATES_B.value] = roc_2_coordinates[3]
+                            roc_3 = roa.roc_3.value
+                            data[CSVFieldnames.ROC_3_NAME.value] = roc_3.name.value
+                            roc_3_coordinates = roc_3.coordinates.value
+                            data[CSVFieldnames.ROC_3_COORDINATES_L.value] = roc_3_coordinates[0]
+                            data[CSVFieldnames.ROC_3_COORDINATES_T.value] = roc_3_coordinates[1]
+                            data[CSVFieldnames.ROC_3_COORDINATES_R.value] = roc_3_coordinates[2]
+                            data[CSVFieldnames.ROC_3_COORDINATES_B.value] = roc_3_coordinates[3]
+                            writer.writerow(data)
+            except Exception as ex:
+                logging.exception(ex)
+                wx.MessageBox(
+                    f"Exporting {pathname} failed, please see log.", "Error", wx.OK | wx.ICON_ERROR
+                )
+
+    def on_close(self, evt):
+        evt.Skip()
+        if not self:
```

Comment:
Can you explain what this does? I think this make sure that if the frame was already destroyed, it doesn't do anything anymore on it.